### PR TITLE
Add SubsetSym to symmetric

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @kortschak 

I'm not sure this is the right name, but I couldn't think of a better one. Documenting the function is a bit tricky even though the behavior is simple. 

This will be used in distmv for computing marginal and conditional distributions of Normal

**Comment by btracey** [comment]
The fast path seems wrong. Not sure why it's passing. Fixing.

**Comment by btracey** [comment]
Fixed. Tests were passing because I forgot to return early so the answer was being overwritten anyway.

**Comment by kortschak** [comment]
I think this need a really clear example. Indexing always does my head in.

**Comment by btracey** [comment]
Example added.

File: mat64/symmetric.go [hunk 1]
```diff
@@ -326,3 +335,42 @@ func (s *SymDense) ScaleSym(f float64, a Symmetric) {
 		}
 	}
 }
+
+// SubsetSym extracts a subset of the rows and columns of the matrix a and stores
+// the result in-place into the receiver. The resulting matrix size is
+// len(set)×len(set). Specifically, at the conclusion of SubsetSym,
+// s.At(i, j) equals a.At(set[i], set[j]). Note that the supplied set does not
```
Comment (kortschak) [inline mat64/symmetric.go]: Add "for all (i, j) of the receiver" or something similar after "a.At....".

File: mat64/symmetric_example_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	s := mat64.NewSymDense(5, nil)
+	count := 1.0
+	for i := 0; i < n; i++ {
+		for j := i; j < n; j++ {
+			s.SetSym(i, j, count)
+			count++
+		}
+	}
+	fmt.Println("Original matrix:")
+	fmt.Printf("%0.4v\n", mat64.Formatted(s))
```
Comment (kortschak) [inline mat64/symmetric_example_test.go]: Wandered off into commit land again.

Extra newline at end of each fmt.Printf format (except last).

File: mat64/symmetric_example_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	// Take the subset {0, 2, 4}
+	var sub mat64.SymDense
+	sub.SubsetSym(s, []int{0, 2, 4})
+
```
Comment (kortschak) [inline mat64/symmetric_example_test.go]: Delete this line.

**Comment by kortschak** [comment]
LGTM with minor comments.

File: mat64/symmetric_example_test.go (hunk 2 above)
Comment (btracey) [inline mat64/symmetric_example_test.go]: Done.

File: mat64/symmetric_example_test.go (hunk 3 above)
Comment (btracey) [inline mat64/symmetric_example_test.go]: Done.

**Pull request merged**